bids/layout/index.py

```diff
@@ -250,6 +250,9 @@ def _index_file(self, abs_fn, entities):
             m = e.match_file(bf)
             if m is None and e.mandatory:
                 break
+            # Skip entities that don't match the filter
+            if self.filters and e.name and e.name in self.filters and m and m not in self.filters[e.name]:
+                continue
             if m is not None:
```

Comment:
This is the equivalent of 

```suggestion
            if m is not None and (e.name not in self.filters or m in listify(self.filters[e.name])):
```